framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceScenes.java

```diff
@@ -452,6 +460,44 @@ public void updateInferredMethodReturnType(
         AnnotatedTypeMirror rhsATM = atf.getAnnotatedType(retNode.getTree().getExpression());
         helper.updateAnnotationSetInScene(
                 method.returnType, atf, jaifPath, rhsATM, lhsATM, TypeUseLocation.RETURN);
+
+        // Now, update overridden methods based on the implementation we just saw.
+        // If the overridden method is abstract, then WPI cannot update it directly
+        // (because no calls directly resolve to it). This inference is similar to
+        // the inference procedure for method parameters: both are updated based
+        // only on the implementations (in this case) or call-sites (for method
+        // parameters) that are available to WPI.
+        for (Map.Entry<AnnotatedDeclaredType, ExecutableElement> pair :
+                overriddenMethods.entrySet()) {
+
+            AnnotatedDeclaredType superclassType = pair.getKey();
+            ExecutableElement superclassMethodDef = pair.getValue();
+
+            AnnotatedExecutableType overriddenMethod =
```

Comment:
It's confusing to use two different prefixes "superclass" and "overridden" for the same concept.

Reply:
You're right about `superclassMethodDef`. However, I think `superclassType` is a reasonable name for that `AnnotatedDeclaredType`, which is the type of the superclass which originally defined the method in question. To make it more clear, I've changed the name to `superclassDecl`.